src/pygama/lgdo/lh5_store.py

```diff
@@ -344,7 +344,19 @@ def read_object(
                 # fields. If they all had shared indexing, they should be in a
                 # table... Maybe should emit a warning? Or allow them to be
                 # dicts keyed by field name?
-                obj_dict[field], _ = self.read_object(
+                if "keys_dtype" in h5f[name].attrs:
```

Comment:
I agree that we don't need to support `bool`s or `float`s, but do you think we need some sort of flag that says whether or not to cast to an `int`? For example, if something was labeled as `'00001'` and you want to use that as a key instead of automatically casting it to the int `1`.

Reply:
Ok, let's add this extra attribute. Can you remove support for `int`s and `float`s?